include/swift/SIL/SILInstruction.h

```diff
@@ -343,8 +373,150 @@ class UnaryInstructionBase : public BASE {
   typename std::enable_if<has_result<X>::value, SILType>::type
   getType() const { return ValueBase::getType(); }
 
-  ArrayRef<Operand> getAllOperands() const { return Operands.asArray(); }
-  MutableArrayRef<Operand> getAllOperands() { return Operands.asArray(); }
+  ArrayRef<Operand> getAllOperands() const {
+    return {&Operands.asArray()[0],
+            static_cast<size_t>(1 + (hasOpenedArchetypesOperands() ? 1 : 0))};
+  }
+
+  MutableArrayRef<Operand> getAllOperands() {
+    return {&Operands.asArray()[0],
+            static_cast<size_t>(1)};
```

Comment:
Why are any of these changes to UnaryInstructionBase necessary?  They seem like detritus from an earlier version of the patch.